packages/graphql-yoga/__tests__/requests.spec.ts

```diff
@@ -0,0 +1,246 @@
+      method: 'GET',
+    })
+    expect(response.status).toBe(404)
+  })
+
+  it('should send basic query', async () => {
+    const response = await yoga.fetch('http://yoga/test-graphql', {
+      method: 'POST',
+      headers: { 'content-type': 'application/json' },
+      body: JSON.stringify({ query: '{ ping }' }),
+    })
+
+    expect(response.status).toBe(200)
+    const body = JSON.parse(await response.text())
+    expect(body.errors).toBeUndefined()
+    expect(body.data.ping).toBe('pong')
+  })
+
+  it('should send basic query with GET', async () => {
+    const response = await yoga.fetch(
+      `http://yoga/test-graphql?query=${encodeURIComponent('{ ping }')}`,
+      {
+        method: 'GET',
+      },
+    )
+
+    expect(response.status).toBe(200)
+    const body = JSON.parse(await response.text())
+    expect(body.errors).toBeUndefined()
+    expect(body.data.ping).toBe('pong')
+  })
+
+  it('sending mutation over GET method is prohibited', async () => {
+    const response = await yoga.fetch(
+      `http://yoga/test-graphql?query=${encodeURIComponent(
+        'mutation { __typename }',
+      )}`,
+      {
+        method: 'GET',
+      },
+    )
+
+    expect(response.status).toBe(405)
+
+    expect(response.headers.get('allow')).toEqual('POST')
+    const body = JSON.parse(await response.text())
+
+    expect(body.data).toEqual(null)
+    expect(body.errors).toHaveLength(1)
+    expect(body.errors[0].message).toEqual(
+      'Can only perform a mutation operation from a POST request.',
+    )
+  })
+
+  it('should send basic mutation', async () => {
+    const response = await yoga.fetch(`http://yoga/test-graphql`, {
+      method: 'POST',
+      headers: { 'content-type': 'application/json' },
+      body: JSON.stringify({
+        query: /* GraphQL */ `
+          mutation {
+            echo
+          }
+        `,
+      }),
+    })
+
+    expect(response.status).toBe(200)
+    const body = JSON.parse(await response.text())
+    expect(body.errors).toBeUndefined()
+    expect(body.data.echo).toBe(null)
+  })
+
+  it('should send variables', async () => {
+    const response = await yoga.fetch(`http://yoga/test-graphql`, {
+      method: 'POST',
+      headers: { 'content-type': 'application/json' },
+      body: JSON.stringify({
+        query: /* GraphQL */ `
+          mutation Foo($text: String!) {
+            echo(str: $text)
+          }
+        `,
+        variables: { text: 'hello' },
+      }),
+    })
+
+    expect(response.status).toBe(200)
+    const body = JSON.parse(await response.text())
+    expect(body.errors).toBeUndefined()
+    expect(body.data.echo).toBe('hello')
+  })
+
+  it('should error on malformed JSON parameters', async () => {
+    const response = await yoga.fetch(`http://yoga/test-graphql`, {
+      method: 'POST',
+      headers: { 'content-type': 'application/json' },
+      body: '{ "query": "{ ping }"',
+    })
+    expect(response.status).toBe(400)
```

Comment:
It seems like this raises a 500 instead of 400 whene xecuted via `yoga.fetch` instead of supertest.

Reply:
And what is the error message?

Reply:
I couldn't find where we throw 400 if request parser fails so probably it was supertest throwing it rather than us. Maybe we should do that internally.

Reply:
https://github.com/dotansimha/graphql-yoga/pull/1567/commits/30ad90bef44337361512ad1c1c1da5d59f4e35e1 is this the correct handling here?